addons/README.md

```diff
@@ -1,25 +1,412 @@
+provided.
+```yaml
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+   name: my-cluster
+   namespace: my-ns
+spec:
+   kapp:
+      refName: kapp-controller.tanzu.vmware.com.0.30.3
+      valuesFrom:
+         providerRef:
+            apiGroup: run.tanzu.vmware.com
+            kind: KappControllerConfig
+            name: test-cluster-tcbt-2-kapp-controller-config
+   additionalPackages:
+      - refName: foobar1.example.com.1.17.2
+        valuesFrom:
+           secretRef: foobar2secret
+   cni:
+      refName: antrea.tanzu.vmware.com.1.2.3--vmware.1-tkg.1
+      valuesFrom:
+         providerRef:
+            apiGroup: cni.tanzu.vmware.com
+            kind: AntreaConfig
+            name: test-cluster-tcbt-2
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+   name: my-cluster
+   namespace: my-ns
+spec:
+   infrastructureRef:
+      kind: VSphereCluster
+   clusterNetwork:
+      pods:
+         cidrBlocks: [ "192.168.0.0/16","fd00:100:96::/48" ]
+      services:
+         cidrBlocks: [ "192.168.0.0/16","fd00:100:96::/48" ]
+...
+...
+```
+
+#### Day 2 customization
+A package and its configuration can be modified post cluster creation by updating associated Config. It will be retained
+post cluster upgrade. Note that package versions are tightly controlled by the TKR, if the version is bumped then it will 
+be reset on a cluster upgrade.
+
+#### Defaulting webhook for ClusterBootstrap
+A defaulting webhook for ClusterBootstrap allows a client to provide partial information and the webhook will fill out 
+rest of the information. The webhook will default whenever it sees an annotation that matches a TKR. The annotation is 
+of the format `tkg.tanzu.vmware.com/add-missing-fields-from-tkr:v1.22.4`.
+
+Additionally a client can provide a wildcard for a package `refName` and the webhook will find the appropriate version based
+on the TKR
+```yaml
+apiVersion: run.tanzu.vmware.com/v1alpha3
+kind: ClusterBootstrap
+metadata:
+   name: my-cluster
+   namespace: my-ns
+   annotations:
+      tkg.tanzu.vmware.com/add-missing-fields-from-tkr:v1.22.4
+spec:
+   cni:
+      refName: calico*
+      valuesFrom:
+         providerRef:
+            apiGroup: cni.tanzu.vmware.com
+            kind: CalicoConfig
+            name: sample-config
+```
+
+## Provider values to a Package
+Configuration for a package can be provided using one of three approaches. The following illustrates all three but only
+one or none can be provided. By definition providerRef and secretRef cannot cross namespace boundary.
+```yaml
+...
+refName: mypackage.acme.com.1.23
+valuesFrom:
+   providerRef:
+      apiGroup: cni.tanzu.vmware.com
+      kind: CalicoConfig
+      name: sample-config
+   secretRef: my-secret-key
+   inline:
+      key1: value1
+      nested_key: 
+        key2: value2
+```
```

Comment:
Thought: Though it is mentioned in the paragraph above, isn't it confusing that the example lists all ```valuesFrom```  options altogether? Wouldn't it be better if this was in the form of 3 different examples?

Reply:
Good point, added 3 separate examples